Migrate kibana overview plugin js to ts Part 1

paulinashakirova opened the pull request:
## Summary

This PR is a part from [[Port Home Page Plugin and Kibana Overview Plugin from JS to TS]](https://github.com/elastic/kibana-team/issues/1307) epic which migrates the last part of the Kibana Overview plugin to `ts`.

- Moved Synopsis component to ts
- Updated unit tests to use testing-library

In the future PR, I will upgrade/change the rest of the tests in this plugin to use testing-library instead of enzyme. (For those components that are already written in ts.)

Commit:
Migrate kibana overview plugin js to ts Part 1

elasticmachine commented:
Pinging @elastic/appex-sharedux (Team:SharedUX)

Dosant commented on src/platform/plugins/private/kibana_overview/public/components/synopsis/synopsis.tsx:
```diff
@@ -0,0 +1,58 @@
+  return (
+    <EuiCard
+      {...(betaBadgeProps && { betaBadgeProps })}
+      className={`homSynopsis__card ${!wrapInPanel ? 'homSynopsis__card--noPanel' : ''}`}
```
are these classes used anywhere?

paulinashakirova replied:
Nice catch!
Everywhere `wrapInPanel` is set to true, so this is unused.
In this case - what do you think about removing this property completely?

paulinashakirova replied:
Based on what we discussed - adding here that styles/classes removal will be done in a separate dedicated to migration from SCSS to Emotion PR.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/276929)
* Commit: dc3e8c32b2857009300a80edded2fe1f6daaabc3

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/276929#01951448-ba8c-4e0d-a0c9-94318180cf08)

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276929","number":276929,"commit":"dc3e8c32b2857009300a80edded2fe1f6daaabc3"}],"number":276929}
buildkite-pr-comment-->

Commit:
remove unused class and add deleted files

Commit:
Merge branch 'main' into js-to-ts-kibana-overview-sharedUX

jedrazb committed:
[Integration] Hide elastic_connectors package (#211419)

## Summary

Hide `elastic_conenctors` package by default. This excludes the package
from:
- integration page
- search results

#### Verification

Without the change package shows up as the integration, and search
results. With the change it's correctly excluded.

Commit:
revert wrapInPanel style class deletion

JoseLuisGJ committed:
[Search] Consuming plugs and web EUI icons and removing custom ones (#211416)

## Summary

`plugs` and `web` new icons are available since `EUI 99.2.0` . As agreed
removing the custom temporary instances and consuming them from EUI
instead.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
Bump backport to v9.6.6 (#211424)

This bumps backport to 9.6.6. Most notably fixes an issue with
Handlebars templates and stripping markdown comments:
https://github.com/sorenlouv/backport/compare/v9.6.4...v9.6.6

Also bumped here:
https://github.com/elastic/kibana-github-actions/pull/55

akowalska622 committed:
[Discover] Allow command/ctrl click for "New" action in top nav (#210982)

## Summary

Closes #198887 

[A-la-carte](https://akowalska622-pr-210982-allow-cmd-click-new-action.kbndev.co/)
to click around

Allows to convert button into anchor tag if `href` is passed to the
element.
Allows to use modifier key to open view in a new tab, without loosing
the current view.


![Cmd+click](https://github.com/user-attachments/assets/90499445-cdc4-4385-8273-2e83267e52e9)


### Checklist

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

jgowdyelastic committed:
[ML] File upload: Adding link to full file upload tool (#211443)

Adds a link to the original file upload page from file upload lite help
text.


![image](https://github.com/user-attachments/assets/12c9799f-95e6-4fa4-81f3-d997a689677a)

Commit:
[Obs AI Assistant] Add KB re-indexing when encountering `semantic_text` bug (#210386)

Closes https://github.com/elastic/kibana/issues/210204

This will automatically re-index the knowledge base if upon adding a KB
entry there is this error:

> The [sparse_vector] field type is not supported on indices created on
versions 8.0 to 8.10

That error means that semantic_text is not supported in the given index,
and it should therefore be re-indexed.

**How to test this PR:**

**8.10**
- `git checkout -B 8.10 origin/8.10`
- Start Kibana:
  - `nvm use && yarn kbn bootstrap && yarn start`
- Start ES
- `nvm use && yarn es snapshot --license trial --E
path.data="/Users/sorenlouv/elastic/kbn_es_data/upgrade_testing"`

**8.19**
- `git checkout -B 8.19 origin/8.x`
- Start Kibana:
  - `nvm use && yarn kbn bootstrap && yarn start`
- Start ES
- `nvm use && yarn es snapshot --license trial --E
path.data="/Users/sorenlouv/elastic/kbn_es_data/upgrade_testing"`
- Install Knowledge base
- Try adding an item to KB (it should fail ❌️)

**9.1.0**
- `gh pr checkout 210386`
- Start Kibana:
  - `nvm use && yarn kbn bootstrap && yarn start`
- Start ES
- `nvm use && yarn es snapshot --license trial --E
path.data="/Users/sorenlouv/elastic/kbn_es_data/upgrade_testing"`
- Try adding an item to KB (it should succeed ✅️)

**TODO:**

- Add an upgrade test that covers this flow

---------

Co-authored-by: Viduni Wickramarachchi <viduni.ushanka@gmail.com>

afharo committed:
[HTTP] Report telemetry of the configured protocol (#211279)

## Summary

Starting with 9.0, we default to `http2` under certain conditions. We
want to report to telemetry how many deployments actively use `http2`.


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[Profiling] Improve finding a function (#210437)

closes https://github.com/elastic/prodfiler/issues/4534

TopN functions:


https://github.com/user-attachments/assets/1b62a50f-3c6f-4cd5-8971-019ca403893c

Diff TopN functions:


https://github.com/user-attachments/assets/0c598317-9423-44b4-9d3f-079d22a0194c

mistic committed:
skip flaky suite (#203607)

elastic-renovate-prod[bot] committed:
Update dependency @redocly/cli to ^1.28.5 (main) (#211283)

kibanamachine committed:
[api-docs] 2025-02-18 Daily api_docs build (#211518)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/987

georgianaonoleata1904 committed:
[ResponseOps][Rules] Move the params of es query and index threshold rule types (#210197)

Connected with https://github.com/elastic/kibana/issues/195188

## Summary

- Moved params of es query rule type to
`@kbn/response-ops-rule-params/es_query` package
- Moved params of index threshold rule type to
`@kbn/response-ops-rule-params/index_threshold` package

The following constants for the es query rule type have been duplicated:
    - MAX_SELECTABLE_SOURCE_FIELDS
    - MAX_SELECTABLE_GROUP_BY_TERMS
    - ES_QUERY_MAX_HITS_PER_EXECUTION

vadimkibana committed:
[ES|QL] Fix field loading when, both, `METADATA` and `LOOKUP JOIN` present (#211375)

## Summary

Closes https://github.com/elastic/kibana/issues/210080

- Constructs field lookup ES|QL query more robustly. Instead of
extracting the text of the first command, now it uses the AST to get the
source and metadata nodes from the `FROM` command and indices from the
`JOIN` command. Then it constructs a new `FROM` query using the `synth`
API.

jillguyonnet committed:
[Fleet] Update instructions for installing Fleet Server (#211495)

## Summary

Closes https://github.com/elastic/kibana/issues/209079

Before:
![Screenshot 2025-02-17 at 18 06
06](https://github.com/user-attachments/assets/7dee41ff-d2c1-44ba-80f5-f2a8d61e4149)

After:
![Screenshot 2025-02-17 at 18 06
49](https://github.com/user-attachments/assets/5384a063-c7b1-475f-96f3-ce06f82b8a1c)

elena-shostak committed:
Upgraded dompurify to 3.2.4 (#211452)

## Summary

Upgraded `dompurify` package to `3.2.4` version

elena-shostak committed:
Upgraded vega to 5.26.0 (#211458)

## Summary

Upgraded vega to `5.26.0`.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

SiddharthMantri committed:
[Session cleanup] Update session index cleanup to check for missing shards (#205744)

Closes https://github.com/elastic/kibana/issues/205146

## Summary
We run a session clean up task that opens a point in time query to try
and delete any older sessions in the session index. We've noticed that
this task fails quite often with the same error
`no_shard_available_action_exception`. On investigating, it's possible
that the point in time query is opened when there are no shards
available for that index. This PR fixes that by checking if the PIT
query fails with 503 bails if it throws the error - allowing the task to
be tried again in the next run of the task manager. We allow for up to
10 failures of the clean up task in succession before logging an error.


### Testing
Unfortunately, there's no reliable way to simulate missing shards
locally. I've added a new integration test config here:

```
node scripts/functional_tests_server.js --config x-pack/test/security_api_integration/session_shard_missing.config.ts
```

This overrides the ES function to return 503 when opening PIT query and
then attempts to assert the result from the task manager.

### Release note
Updates session cleanup mechanism to account for potential missing
shards in Session index.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.


- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
[kibana-flaky-test-suite-runner#7836](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7836)
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elastic-renovate-prod[bot] committed:
Update OpenFeature (main) (#200273)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [@openfeature/server-sdk](https://togithub.com/open-feature/js-sdk) |
dependencies | patch | [`^1.16.1` ->
`^1.16.2`](https://renovatebot.com/diffs/npm/@openfeature%2fserver-sdk/1.16.1/1.16.2)
|
| [@openfeature/web-sdk](https://togithub.com/open-feature/js-sdk) |
dependencies | patch | [`^1.3.1` ->
`^1.3.2`](https://renovatebot.com/diffs/npm/@openfeature%2fweb-sdk/1.3.1/1.3.2)
|

---

### Release Notes

<details>
<summary>open-feature/js-sdk (@&#8203;openfeature/server-sdk)</summary>

###
[`v1.16.2`](https://togithub.com/open-feature/js-sdk/compare/0cc2590d02ade510a814b02da9eac17aff25072e...1ba149d8e5f56608e87fc10326d1a6acdbcedc05)

[Compare
Source](https://togithub.com/open-feature/js-sdk/compare/0cc2590d02ade510a814b02da9eac17aff25072e...1ba149d8e5f56608e87fc10326d1a6acdbcedc05)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config help](https://togithub.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy40MjUuMSIsInVwZGF0ZWRJblZlciI6IjM3LjQyNS4xIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOkNvcmUiLCJUZWFtOlNlY3VyaXR5IiwiYmFja3BvcnQ6cHJldi1taW5vciIsInJlbGVhc2Vfbm90ZTpza2lwIl19-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

tkajtoch committed:
Fix `SolutionNav` padding in pages that use overridden EUI breakpoints (#211513)

elena-shostak committed:
Upgraded octokit to 21.1.1 (#211450)

## Summary

Upgraded octokit to `21.1.1`

juliaElastic committed:
[Fleet] show auto upgrade agent status (#210866)

## Summary

Closes https://github.com/elastic/ingest-dev/issues/4731

Added API to query agents per each version enrolled to an agent policy,
including the count of agents that failed to upgrade to the target
version.
This API is used on the UI to calculate the status of auto upgrade
status.
- Complete status: agent count reaches or exceeds target percentage
without upgrade failures
- Not started status: 0 agents on target percentage
- In progress status: agent count doesn't reach target percentage
- Failed status: there is at least one agent in failed upgrade status on
the target version

Added click handler to navigate from the status to agent list.

[UI
Design](https://www.figma.com/design/ZH58ySPR1nhI3lRHrkds1t/%5BFleet%5D-Automatic-target-agent-version?node-id=2128-4286&p=f&t=3GvdUYkqdCAJj99q-0)



```
GET kbn:/api/fleet/agent_policies/{agentPolicyId}/auto_upgrade_agents_status

{
  "currentVersions": [
    {
      "version": "8.16.2",
      "agents": 1,
      "failedAgents": 0
    },
    {
      "version": "8.16.3",
      "agents": 0,
      "failedAgents": 1
    }
  ],
  "totalAgents": 1
}
```

<img width="985" alt="image"
src="https://github.com/user-attachments/assets/c4a40872-2fe7-48e9-ade5-fe2b36d06a0e"
/>
<img width="811" alt="image"
src="https://github.com/user-attachments/assets/1cbbebb9-2c7b-4cb5-a57d-7a0388fd437e"
/>
<img width="805" alt="image"
src="https://github.com/user-attachments/assets/be716b87-9b0f-4f16-ad05-a463169bee25"
/>
<img width="989" alt="image"
src="https://github.com/user-attachments/assets/09745005-d5ed-4a80-8e7a-7c2df427d4ad"
/>
<img width="906" alt="image"
src="https://github.com/user-attachments/assets/162d43aa-1bce-4dda-a5dd-91cb59636740"
/>


### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

umbopepato committed:
[ResponseOps][Observability][Alerts] Fix missing alert grouping controls in o11y alerts page (#211160)

## Summary

Adds back the additional alerts table toolbar controls to edit the
grouping configuration. Adds test cases to check the correctness of the
Observability alerts table configurations.

## To verify

1. Create one or more rules that fire alerts in Observability
2. Navigate to Observability > Alerts
3. Verify that the grouping toggle shows and works correctly in the
table toolbar (`Group by: ...`)

Commit:
[Obs AI Assistant] Defer KB migration until index assets have been updated (#211523)

Follow-up to https://github.com/elastic/kibana/pull/210386
Closes: https://github.com/elastic/kibana/issues/211525

This ensures that we don't start executing the knowledge base migration
until index assets have been updated

kowalczyk-krzysztof committed:
[Spaces] Show callout on spaces permission tab when security is disabled (#210961)

## Summary

This PR updates the behavior of `Permissions` tab in `Space Management`
when `xpack.security.enabled` is set to `false` to show a callout with a
meaningful explanation.
![Screenshot 2025-02-17 at 12 42
58](https://github.com/user-attachments/assets/881d0bbe-e30b-41b4-8c0a-5b8e127786ab)


Closes: #210241

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

efegurkan committed:
[Fix]Removes synonyms read-only permission. (#211471)

## Summary

Kibana permission for read-only is removed. This is not a breaking
change while the feature is not yet released.
Cluster requirements make it obsolete.
Also fixed warnings on permission names.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

mbondyra committed:
[Lens] Remove scss from annotations plugin, visualization-ui-components and gauge expression (#208891)

## Summary

part of https://github.com/elastic/kibana/issues/208908

Replaces scss to css-in-js. I've tested all the changes.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
🌊 Move kbn-streams-schema to platform (#211230)

First part of moving streams into the platform. This PR moves the
package only, will stack PRs on top of this.

alexreal1314 committed:
10973 migrate flaky e2e tests to jest 2 (#211363)

## Summary

This PR tries to fix the following issues - which are flaky FTR tests:

- https://github.com/elastic/kibana/issues/209529
- https://github.com/elastic/kibana/issues/201686
- https://github.com/elastic/kibana/issues/203680
- https://github.com/elastic/kibana/issues/178413 - test is still
skipped, added unit tests for rules table, header and counter
components.
- https://github.com/elastic/kibana/issues/193073
- https://github.com/elastic/kibana/issues/193616
- https://github.com/elastic/kibana/issues/191604
- https://github.com/elastic/kibana/issues/191593
- https://github.com/elastic/kibana/issues/191511
- https://github.com/elastic/kibana/issues/191474
- https://github.com/elastic/kibana/issues/191322
- https://github.com/elastic/kibana/issues/191128
- https://github.com/elastic/kibana/issues/191144
- https://github.com/elastic/kibana/issues/191027
- https://github.com/elastic/kibana/issues/190831
- https://github.com/elastic/kibana/issues/190779



There will be an RFC document which is going to be released to help us
better understand and decide which tests are more suitable to make as
E2E tests and which as unit tests.

### Checklist
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

### Closes
this PR closes the above mentioned issues in relation for this ticket -
https://github.com/elastic/security-team/issues/10973

nchaulet committed:
[Fleet] Add UI to add additional datastreams permissions (#210935)

nchaulet committed:
[Fleet] Fix flaky agent status field (#211453)

## Summary

Resolve https://github.com/elastic/kibana/issues/209008 

It seems with ES > 9 that runtime field is sometimes failing, while I am
not sure why, that PR make it more robust and should avoid test
flakyness.

---------

Co-authored-by: Julia Bardi <90178898+juliaElastic@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Samiul-TheSoccerFan committed:
[Search] Adding tech preview for sprinkles model (#211519)

## Summary

Adding `Tech Preview` for `rainbow sprinkles` model


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [X] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

kdelemme committed:
chore(slo): link to SLO details page from create and update toast (#211475)

CohenIdo committed:
Deprecate universal entity

meghanmurphy1 committed:
[Search Connectors] : Unregister Kibana background task (#211091)

## Summary

Since Agentless Connectors will not be released in 9.0 anymore, this PR
should unregister and stop the background kibana task from creating new
policies for elastic-managed connectors.

### Closes https://github.com/elastic/search-team/issues/9283

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#210245)

juliaElastic committed:
[Fleet] Added warning icon next to auto upgrade agents badge (#211551)

## Summary

Closes https://github.com/elastic/ingest-dev/issues/4731

Added warning icon to Agent policy details, next to Auto upgrade agents
badge if there are failures in agents upgrade.

UX Design:
<img width="1421" alt="image"
src="https://github.com/user-attachments/assets/a2c54fef-0100-48ed-9461-eb9f8c704a3d"
/>

UI:
<img width="1571" alt="image"
src="https://github.com/user-attachments/assets/ee061feb-c489-4637-b684-f489c822aab9"
/>

flash1293 committed:
Move useAbortController and useAbortableAsync hook into central package (#211295)

The useAbortController and useAbortableAsync hooks are very generic
hooks that are useful beyond observability. This PR moves them into
`react-hooks` which already houses other similar hooks.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[visualize] fix unsaved state when adding by-value visualize embeddable to dashboard (#211264)

Follow up to https://github.com/elastic/kibana/pull/210125

[8.16](https://github.com/elastic/kibana/pull/211057) and
[8.17](https://github.com/elastic/kibana/pull/211054) backports for
https://github.com/elastic/kibana/pull/210125 were failing functional
test
https://github.com/elastic/kibana/blob/8.17/test/functional/apps/dashboard/group1/dashboard_unsaved_listing.ts#L142.
The functional test adds a by-value and by-reference legacy
visualization to a new dashboard. Upon saving the dashboard, the
dashboard still showed unsaved changes.

The reason this test did not fail main and other branches is that
https://github.com/elastic/kibana/pull/208116 removed the "by-value"
part of the test (since its no longer possible to add a by-value legacy
visualization from within a dashboard). It is still possible to recreate
the issue in main with the following steps
1) Click "Visualize Library" in left nav
2) Click "Create visualization" button. 
3) Click "Legacy" tab
4) Click "Aggregation based"
5) Click "Area"
6) Click web logs sample data view
7) Click "Save"
8) Set title
9) Under "Add to dashboard", click "New", click save
10) save dashboard. Notice how dashboard still has unsaved changes.

8.16 and 8.17 required a [new
commit](https://github.com/elastic/kibana/pull/211054/commits/1fd631c5a30046b5ab2f63948174ec29bac6fd84)
to resolve the issue by updating the `linkedToLibrary` to ignore
undefined values.

This PR fixes the issue for the other branches that have already been
merged.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

rlueder committed:
fix: add REACT_18 env variable to kbn-plugin-helpers bazel config (#211121)

## Summary

Adds `REACT_18` to `kbn-plugin-helpers` bazel config, this fixes an
issue when trying to create plugins using the automatic plugin
generator. For context see
https://github.com/elastic/kibana/pull/210917#issuecomment-2657805726

Related user report: [Custom Plugin
Error](https://discuss.elastic.co/t/custom-plugin-error/366589)

Before fix:


![image](https://github.com/user-attachments/assets/e7e9e524-719e-4fdb-8598-050ec9f2f036)


After fix:


![image](https://github.com/user-attachments/assets/63d69d1a-bfc8-4be6-83a0-1c567be4d101)


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Co-authored-by: Jon <jon@elastic.co>

paul-tavares committed:
[Security Solution][Endpoint] Add the space ID to endpoint artifacts when creating or updating them (#210698)

## Summary

Changes in this PR are in associated with Endpoint Management support
for spaces:

- When creating an endpoint artifact (Trusted Apps, Event Filters,
Blocklists, Host Isolation Exceptions, Endpoint Exceptions), the API
will ensure a new `tag` is dded to each item created that identifies the
space ID from where it was crated
- This functionality is behind the following feature flag:
`endpointManagementSpaceAwarenessEnabled`
- The tag that will be automatically added has a format of:
`ownerSpaceId:<space_id_here>`
- Likewise, when updating an artifact, the API will ensure that at least
1 owner space id tag is present on the item, and if not, it will add one
to it.

nreese committed:
unskip Failing test: X-Pack API Integration Tests.x-pack/test/api_integration/apis/maps/search·ts (#211304)

Closes https://github.com/elastic/kibana/issues/208138

ES|QL response updated with `is_partial` key.
x-pack/test/api_integration/apis/maps/search.ts tests where skipped to
unblock ES-snapshot promotion. This PR unskips the tests and updates the
expects for the new response shape

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

eokoneyo committed:
Consolidate all language definitions used in monaco within the @kbn/monaco package (#208950)

## Summary

This PR was created in response to
https://github.com/elastic/kibana/pull/208858, it migrates all existing
language definitions within the `@kbn/code-editor` package into the
`@kbn/monaco` package to provide a separation of concern for logic that
doesn't particularly relate to the configurations for the code editor
UI.

With this change, all supported languages are ingested from the
`@kbn/monaco` package where they will be domiciled from henceforth, and
in turn fix the issue that was discovered relating to the way the
language definitions within `@kbn/code-editor` get registered as a side
effect.

With this change, to add support for a new language, said language
should be defined within `@kbn/monaco`, and registered in
`languages/index.ts`.

<!--
### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

-->

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elastic-renovate-prod[bot] committed:
Update ftr (main) (#208934)

This PR contains the following updates:

| Package | Type | Update | Change | Pending |
|---|---|---|---|---|
| [chromedriver](https://redirect.github.com/giggio/node-chromedriver) |
devDependencies | major | [`^132.0.0` ->
`^133.0.0`](https://renovatebot.com/diffs/npm/chromedriver/132.0.1/133.0.0)
| `133.0.1` |
|
[selenium-webdriver](https://redirect.github.com/SeleniumHQ/selenium/tree/trunk/javascript/node/selenium-webdriver#readme)
([source](https://redirect.github.com/SeleniumHQ/selenium)) |
devDependencies | patch | [`^4.28.0` ->
`^4.28.1`](https://renovatebot.com/diffs/npm/selenium-webdriver/4.28.1/4.28.1)
| |

---

### Release Notes

<details>
<summary>giggio/node-chromedriver (chromedriver)</summary>

###
[`v133.0.0`](https://redirect.github.com/giggio/node-chromedriver/compare/132.0.2...133.0.0)

[Compare
Source](https://redirect.github.com/giggio/node-chromedriver/compare/132.0.2...133.0.0)

###
[`v132.0.2`](https://redirect.github.com/giggio/node-chromedriver/compare/132.0.1...132.0.2)

[Compare
Source](https://redirect.github.com/giggio/node-chromedriver/compare/132.0.1...132.0.2)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://redirect.github.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy40MjUuMSIsInVwZGF0ZWRJblZlciI6IjM5LjEwNy4wIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOk9wZXJhdGlvbnMiLCJiYWNrcG9ydDpza2lwIiwicmVsZWFzZV9ub3RlOnNraXAiXX0=-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>
Co-authored-by: Jon <jon@elastic.co>

andrew-goldstein committed:
[Security Solution] [Attack discovery] Fixes alerts filtering issues (#211371)

### [Security Solution] [Attack discovery] Fixes alerts filtering issues

This PR resolves the following Attack discovery alerts filtering issues:

- [[Security Solution] [Bug] A few filters show error 'Unexpected error from Elasticsearch' for the alerts flyout #208481](https://github.com/elastic/kibana/issues/208481)
- [[Security Solution] [Bug] Lucene not updated as the space holder when we apply Lucene as the filtering language #208170](https://github.com/elastic/kibana/issues/208170)
- Connector selection resets in non-default spaces
- Saving a filter edited via `Edit Query DSL` with an unknown `user.name` value results in a `filter value is invalid or incomplete` filter
- Local field Reset clears the preview dropdowns when they are in an error state
- Updates the formatting of `Up to _n_ alerts` for Borealis

#### Feature flag required for testing

The following feature flag is required to test this PR:

```yaml
xpack.securitySolution.enableExperimental:
  - 'attackDiscoveryAlertFiltering'
```

The following sections provide details and desk testing steps for the alerts filtering issues fixed by this PR.

### [Security Solution] [Bug] A few filters show error 'Unexpected error from Elasticsearch' for the alerts flyout #208481

To resolve [[Security Solution] [Bug] A few filters show error 'Unexpected error from Elasticsearch' for the alerts flyout #208481](https://github.com/elastic/kibana/issues/208481):

- The `_ignored` metadata field was added to the [METADATA](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-metadata-fields.html) keyword in the `ES|QL` queries that power the `Alert summary` and `Alerts preview` tabs

- The `Alert summary` and `Alerts preview` tabs dropdowns are now filtered to only display fields in the alerts index
  - Example: Previously, if the `dll.Ext.load_index` field was in a `logs-*` index, but not the alerts index, it would still appear in the `Alerts summary` dropdown, and result in an error if selected. After the fix in this PR, this (example) field would not appear in the dropdown.

#### Desk testing

To desk test this fix:

- See issue <https://github.com/elastic/kibana/issues/208481>

- In addition to the steps in [issue #208481](https://github.com/elastic/kibana/issues/208481), verify the `Alert summary` and `Alerts preview` dropdowns
  - Do NOT display an error when the `_ignored` field is selected
  - Identify a field that is present in a `logs-*` index, but not the alerts index, for example `dll.Ext.load_index`. Verify this field is NOT displayed in the dropdown

### [Security Solution] [Bug] Lucene not updated as the space holder when we apply Lucene as the filtering language #208170

To resolve [[Security Solution] [Bug] Lucene not updated as the space holder when we apply Lucene as the filtering language #208170](https://github.com/elastic/kibana/issues/208170), the custom placeholder was removed, as illustrated by the _before_ and _after_ screenshots below:

**Before**

![settings before](https://github.com/user-attachments/assets/4bab48bd-e0b5-42eb-93fe-3faefdfc58bf)

**After**

![settings after](https://github.com/user-attachments/assets/b499dab0-0ee1-464a-8bda-cdbf5236b0d3)

#### Desk testing

To desk test this fix, see <https://github.com/elastic/kibana/issues/208170>

### Connector selection resets in non-default spaces

This PR fixes an issue where the last selected connector would reset in non-default spaces when all of the following were true:

- The user is in a non-default space
- Two or more generative AI connectors are configured

This issue occurred in non-default spaces because:

- `spaceId` loads asynchronously
- Kibana's `package.json` references an older version of `react-use`, with a known bug in the `useLocalStorage` hook, which is fixed by <https://github.com/streamich/react-use/pull/1944>
- I verified (locally) the fix from <https://github.com/streamich/react-use/pull/1944> would work if Kibana's version of `react-use` was updated in `package.json`, however that effort appears to be on hold: <https://github.com/elastic/kibana/pull/179268> . For now (to minimize changes), `spaceId` has been removed from all Attack discovery local storage keys.

#### Desk testing

1. Create a new space (if you only have the default space)

2. Configure two or more Gen AI connectors

3. Select the newly created space

4. Navigate to Security > Attack discovery

5. Select a connector, for example `Claude 3.5 Sonnet`

6. Now select a _different_ connector, for example `Gemini 1.5 Pro 002`

7. Navigate to a different page in the Security solution, for example Security > Alerts

9. Once again, navigate to Security > Attack discovery

**Expected result**

- The previously selected connector, e.g. `Gemini 1.5 Pro 002` is still selected

### Saving a filter edited via `Edit Query DSL` with an unknown `user.name` value results in a `filter value is invalid or incomplete` filter

This PR fixes an issue where editing a previously created non-Query DSL filter via `Edit Query DSL`, and then entering an unknown `user.name`, resulted in a filter with text that reads: `filter value is invalid or incomplete`, as illustrated by the screenshot below:

![filter value is invalid](https://github.com/user-attachments/assets/39493dba-bf1d-4ce7-8480-15ee2ed599ea)

Generating attack discoveries with a filter like the one shown in the screenshot above would also result in errors.

This issue was resolved by adding a `FilterManager` to manage the local state of the filters in the settings panel.

#### Desk testing

1. Navigate to Security > Attack discovery

2. Click the settings gear

3. Click the `+` button to open the `Add filter` popover

4. In the popover, configure a `user.name` `is` `Administrator` filter

  Note: replace `Administrator` with a real `user.name`  value if your alerts index doesn't have the value ``Administrator``

5. Click `Add filter` to close the popover

**Expected result**

- The `user.name: Administrator` filter appears below the query bar

6. Click the `user.name: Administrator` filter, and choose `Edit filter` from the popover

7. Click the `Edit as Query DSL` button (in the upper right hand corner)

8. In the `Edit filter` Elasticsearch Query DSL editor, edit the Query DSL such that it has a value that does NOT exist in the index, like the following example:

```json
{
  "match_phrase": {
    "user.name": "Admasdfinistrator"
  }
}
```

9. Click `Update filter`

**Expected results**

- The `user.name: Admasdfinistrator` filter, which references a value that does not exist in the alerts index, appears below the query bar
- The updated filter does NOT have the text `filter value is invalid or incomplete`, as illustrated by the the screenshot in the description of this issue above.

### Local field Reset clears the preview dropdowns when they are in an error state

This PR fixes an issue where the local (to the preview tab) reset button did not clear the preview dropdowns if they were in an error state.

The issue is fixed by calling `clearSearchValue()` to reset the stack by field when it's in an error state (i.e. because an invalid field was entered)

Note: The "local" (to the tab) `Reset` button shown in the screenshot below is fixed by this PR:

![local_reset](https://github.com/user-attachments/assets/0a2d040f-c31a-40b0-8c16-04b7d333f73e)

, however the `Reset` button at the bottom of the flyout will NOT clear the dropdown if it's in an error state. (For now, this is the expected behavior.) The workaround is to manually select a valid value in the dropdown, or click `Save` or `Cancel`. (The preview dropdown does not effect the Attack discovery query, is not saved, and automatically resets to the default every time the flyout is opened.)

#### Desk testing

1. Navigate to Security > Attack discovery

2. Click the settings gear

3. In the `Alert summary` tab, focus the dropdown and delete the text until it reads:

```
kibana.alert.rule.na
```

4. Blur the dropdown by clicking outside it

**Expected results**

- The dropdown is highlighted red
- The `Reset` button appears below the text `Select a field`

5. Click the `Reset` button below the text `Select a field`

**Expected results**

- The dropdown is NOT highlighted red (the error state is cleared)
- The dropdown text is reset to the (valid) default value: `kibana.alert.rule.name`

### Updates the formatting of `Up to _n_ alerts` for Borealis

This PR updates the formatting of the `Up to n alerts` text in Borealis, as illustrated by the before and after screenshots below:

**Before**
![01-up_to_100_alerts_before](https://github.com/user-attachments/assets/4143e847-5220-463b-8fb0-da5215d16b24)

**After**
![02-up_to_100_alerts_after](https://github.com/user-attachments/assets/835bd3fb-1e63-4192-b694-4595e8fa9309)

#### Desk testing

To desk test this fix:

1. Configure Kibana to use the `dark` theme

2. Navigate to Security > Attack discovery

3. In the connector selector, choose `+ Add new Connector...`

4. Click in the `Select a connector` dialog, click `OpenAI`

4. Enter a throwaway configuration for the connector (note: you won't actually use it), and click `Save`

**Expected results**

- The animated `Up to 100 alerts will be analyzed` message will appear
- The color of the animated numeric text, e.g. `100` matches the color of the `Up to` text that precedes it
- The extra whitespace trailing the `100`, shown in the _Before_ image (above) does NOT appear. The trailing whitespace after the `100` looks like the _After_ image (also above).

jbudz committed:
Fix octokit types (#211643)

Related to https://github.com/elastic/kibana/pull/211450

banderror committed:
[Security Solution] Refactor prebuilt rule customization test plans, part 2 (#211472)

**Epic:** https://github.com/elastic/kibana/issues/174168
**Partially addresses:**
https://github.com/elastic/kibana/issues/202068,
https://github.com/elastic/kibana/issues/202078,
https://github.com/elastic/kibana/issues/202079
**Follow-up to:** https://github.com/elastic/kibana/pull/211300

## Summary

We're cleaning up and refactoring our existing test plans for prebuilt
rule customization, upgrade, and export/import workflows.

Specifically, this PR:

- Creates an "entrypoint" file that should help navigate all the test
plans for prebuilt rules.
- Creates a file for keeping common information about prebuilt rules
that can be shared between the test plans.
- Extracts duplicated terminology to the file with common information.
- Extracts duplicated assumptions to the file with common information.
- Extracts duplicated non-functional requirements to the file with
common information.
- Adds user stories to each test plan.
- Updates links to tickets in every test plan.

No "functional" changes have been made to any test plans, such as
adding, removing, or updating any scenarios.

This refactoring prepares the test plans for being "functionally"
changed and improved in follow-up PRs. For example, we're going to cover
the logic described in https://github.com/elastic/kibana/issues/210358
and address any other gaps in coverage.

## Review tip

It might be easier to review this PR commit-by-commit as each of them
contains logically cohesive changes.

nreese committed:
[canvas] remove usage of EmbeddableInput type (#211512)

Part of effort to remove legacy EmbeddableInput type.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nchaulet committed:
[Fleet] Make status runtime query more robust (#211639)

PhilippeOberti committed:
[Security Solution][Expandable flyout] show last open in the history (#210286)

## Summary

This PR makes a small UI addition to the expandable flyout package
history functionality [merged not long
ago](https://github.com/elastic/kibana/pull/184970).

We are snow saving the time at which a panel was opened, along side the
panel's information. This allows us to show when previous flyout panels
were opened, making it users easier for users to know which
host/user/network/alerts/event document they recently looked at.

The primary logic change happens in the expandable flyout package. Most
of the other code changes are in Security Solution unit tests.

#### Before

![Screenshot 2025-02-07 at 5 07
48 PM](https://github.com/user-attachments/assets/8f0b79b7-a3db-4d01-8d3c-a4a9b632e9cf)

#### After

![Screenshot 2025-02-13 at 11 03
31 AM](https://github.com/user-attachments/assets/64cd5df2-a0b6-4eee-8555-53aedda1db8f)

You'll notice in the screenshot above that another small improvement was
added: we're limiting the width of the text rendered and show an
ellipsis. This will prevent really long texts to have the popover really
wide or to be rendered over multiple lines

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

davismcphee committed:
[Discover] Fix failing tests due to log level badge colour changes in Borealis (#211329)

## Summary

This PR fixes functional tests that started failing due to changes to
log level badge colours in Borealis. I'm not sure why these started
failing or how the PR where it started passed CI, but based on the
timeline and test file changes, it looks like it's related to #210468.

cc @tkajtoch Maybe you have some idea what's going on here?

Resolves #211174.
Resolves #211215.

### Checklist

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Co-authored-by: Matthias Wilhelm <matthias.wilhelm@elastic.co>

Ikuni17 committed:
Remove SCSS from plugin generator (#211116)

## Summary

Part of elastic/kibana-team#1417, removes `index.scss` from plugin
generation.

kibanamachine committed:
[api-docs] 2025-02-19 Daily api_docs build (#211680)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/988

guskovaue committed:
[ResponseOps] Add Copy Id button for MW (#209135)

Resolve: https://github.com/elastic/kibana/issues/203569

Solution looks like:

https://github.com/user-attachments/assets/424c6518-289b-4119-b909-4e589a618069

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

gsoldevila committed:
SKA: Unify `kbn-management` folders + storybook config (#211564)

## Summary

As part of the SKA, some groups of modules have been broken-down and
relocated in different folders, negatively impacting DX. This is the
case for `kbn-management` modules, that are spread across:
```
- packages/kbn-management (storybook config)
- src/platform/packages/private/kbn-management
- src/platform/packages/shared/kbn-management
```

The goal of this PR is to unify them back under
`src/platform/packages/shared/kbn-management`.
In this scenario, I believe the simplification is worth the trade off.

CC @elastic/kibana-management WDYT?

Co-authored-by: Matthew Kime <matt@mattki.me>

gsoldevila committed:
SKA: Unify `serverless` modules + storybook config (#211570)

## Summary

As part of the SKA, some groups of modules have been broken-down and
relocated in different folders, negatively impacting DX. This is the
case for `serverless` modules, that are spread across:
```
- packages/serverless  # storybook config
- src/platform/packages/private/serverless
- src/platform/packages/shared/serverless
```

The goal of this PR is to unify them back under
`src/platform/packages/shared/severless`.
In this scenario, I believe the simplification is worth the trade off.

CC @elastic/kibana-management @elastic/appex-sharedux WDYT?

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

MiriamAparicio committed:
[ObsUX][Profiling] Remove float numbers on samples (#211489)

Closes https://github.com/elastic/kibana/issues/181295

#### Summary

There should be no decimal / fraction in the number of samples (those
are integer values).

#### What was done

When the scale factor was a number with decimals, was changing the value
of the samples to a float number
Also, the background color for the `Normalized By` dropdown was fixed in
dark mode

Before

<img width="1335" alt="image"
src="https://github.com/user-attachments/assets/93b6c301-5b7f-41b6-857b-3551cf88b095"
/>

After

<img width="1335" alt="image"
src="https://github.com/user-attachments/assets/ebbe78ed-d8cb-4308-acf4-0d15221f986b"
/>

Commit:
update snapshots to match component props change

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/277584)
* Commit: e1a3188bac87f2e2900282fea218b430c744a06a

### Failed CI Steps
* [Osquery Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/277584#01951d9f-997b-43dd-a410-d622972234aa)

### History
* :broken_heart: [Build #276989](https://buildkite.com/elastic/kibana-pull-request/builds/276989) failed 6fb78b90787547210f1933116bf1480616f08e25
* :broken_heart: [Build #276929](https://buildkite.com/elastic/kibana-pull-request/builds/276929) failed dc3e8c32b2857009300a80edded2fe1f6daaabc3

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/277584","number":277584,"commit":"e1a3188bac87f2e2900282fea218b430c744a06a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276989","number":276989,"commit":"6fb78b90787547210f1933116bf1480616f08e25"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276929","number":276929,"commit":"dc3e8c32b2857009300a80edded2fe1f6daaabc3"}],"number":277584}
buildkite-pr-comment-->
